### ISSUE TITLE
Negative overlap features (feature normalization)

### ISSUE THREAD

**Issue by rosequ** [castorini/data#23]
I ran the fixed version of overlap_features.py [code](https://github.com/castorini/Castor/blob/master/sm_cnn/external_features.py#L52) from castor/data, which is supposed to calculate the word-overlap and idf-weighted word overlap. The resulting values should be positive real numbers but this is not really the case:
<img width="521" alt="screen shot 2017-09-19 at 3 17 07 pm" src="https://user-images.githubusercontent.com/8274617/30611114-6d48007a-9d4e-11e7-96dc-6ead42117fc5.png">

**Comment by hatianzhang** [comment]
https://github.com/castorini/data/blob/master/TrecQA/overlap_features.py#L168

As I mentioned, the overlap features is normalized for each dimension. And the function is defined here: 
http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html#sklearn.preprocessing.StandardScaler.fit_transform

**Comment by rosequ** [comment]
@HTAustin I did look at this. The paper, however, doesn't normalize the features. The implementation should closely follow the original paper. Correct me if I'm wrong.

**Comment by hatianzhang** [comment]
I found that the original DeepQA repo commented the normalization code :https://github.com/aseveryn/deep-qa/blob/249a1ec14ef980a5630b1a2dccf5f587245e16e1/run_nnet.py#L79 

@Jeffyrao Torch version used the normalization: https://github.com/castorini/SM-CNN-Torch/blob/master/scripts/overlap_features.py#L158

If we follow the original paper, we might need to remove the normalization. But the good solution would be trying to not using normalization and see whether results improves or drops.

**Comment by rosequ** [comment]
Closing this since I get comparable numbers without normalizing the features:

MAP with and without normalizing:

Metric |static|non-static|multichannel
-------|-------|------|----------
Normalization    |0.7524|0.7688    |0.7641
No normalization| 0.7628|0.7590|0.7764
